src/memray/_memray/tracking_api.cpp

```diff
@@ -1219,6 +1219,16 @@ Tracker::handleGreenletSwitch(PyObject* from, PyObject* to)
     // Grab the Tracker lock, as this may need to write pushes/pops.
     std::unique_lock<std::mutex> lock(*s_mutex);
     RecursionGuard guard;
+
+    // Check if the trace function is still installed in the current thread as
+    // it's possible that the thread is dying and the trace function has been
+    // uninstalled and therefore the stack will not be properly tracked so we
+    // should not recreate the current thread stack.
```

Comment:
This is more explanation than we need, I think. I'd just say:
```suggestion
    // We must stop tracking the stack once our trace function is uninstalled.
```
If you think it deserves more than that, maybe one more line:
```cpp
    // Otherwise, we'd keep referencing frames after they've been destroyed.
```